builder/replica_pipeline.cc

```diff
@@ -146,8 +146,29 @@ class pipeline_replicator : public expr_visitor {
     buffers_emitted_.insert(bep->sym());
 
     std::string elem_size = print_expr_inlined(bep->elem_size());
-    (void)print_assignment_explicit(
-        name, "buffer_expr::make(ctx, \"", name, "\", /*rank=*/", bep->rank(), ", /*elem_size=*/", elem_size, ")");
+    if (bep->constant()) {
+      auto c = bep->constant();
+      std::string const_name = name + "_const";
+      (void)print_assignment_explicit(
+          const_name, "std::make_shared<buffer<void, ", c->rank, ">>(/*rank=*/", c->rank, ", /*elem_size=*/", elem_size, ")");
+      for (std::size_t d = 0; d < c->rank; d++) {
+          std::string mn = print_expr_inlined(c->dim(d).min());
```

Comment:
These are constants, so `print_expr_inlined` is overkill here, you could just `<<` these directly below. I was going to say it doesn't matter, but I think it kinda does: this makes it *look* like these are exprs and not constants.

Reply:
So they are! PTAL